lib/tree.rb

```diff
@@ -16,40 +16,105 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(logn) in worst case scenario
+  # Space Complexity: O(1) always just adding 1 new node
   def add(key, value)
```

Comment:
Actually O(log n) is the **best** and **average** case runtime.

Since you're doing recursion, you do incur some space complexity.

Reply:
@CheezItMan for the time complexity - would it be O(n) for worst case, and is that possible only if the tree is unbalanced?